src/index.js

```diff
@@ -179,9 +179,9 @@ class Jimp extends EventEmitter {
 
             setTimeout(() => {
                 // run on next tick.
-                if (err) {
+                if (err && cb === noop) {
                     jimpInstance.emitError('constructor', err);
-                } else {
```

Comment:
I am not sure why you want to change the behavior of this ?

Reply:
(The rest looks good to me)

Reply:
If jimp emits the error then the callback will never be called with error, and the promise never catches the error. this change makes it so that jimp either 

1. calls the callback with the error
2. rejects with the error
3 or if no callback or promise is provided it emits the error

when we were calling emitError the whole execturion of the program would stop if you didn't have an `on('error')`. All the issues basically arose from just this.